src/__tests__/mockStates.js

```diff
@@ -1219,10 +1221,40 @@ export const rolesMock = {
           'editTeamCode',
         ],
       },
+      {
+        roleName: 'Fake Test Role',
+        permissions: [],
+      },
     ]
   }
 }
 
 describe('Stop Error', () => {
   it('should not error out due to no tests (mockStates.js)', () => {});
 });
+
+// takes a list of permissions and returns a list of all other permissions
+const allPermissionsExcept = (permissions) => {
+  return getAllPermissionKeys().filter(perm => !permissions.includes(perm))
+}
+
+// takes a list of relevant permissions and returns two auth objects, one with the permissions and the other with all permissions not listed
+export const createAuthMocks = (permissions) => {
+  var authTemplate = {
+    // isAdmin: true,
+    user: {
+      userid: '5edf141c78f1380017b829a6',
+      role: 'Fake Test Role',
+      expiryTimestamp: '2020-08-22T22:51:06.544Z',
+      iat: 1597272666,
+    },
+    permissions: {
+      frontPermissions: []
+    },
+    firstName: 'Dev',
+    profilePic:''
+  };
+  const onlyPermissions = {...authTemplate, permissions: {frontPermissions: permissions}};
+  const allOtherPermissions = {...authTemplate, permissions: {frontPermissions: [...allPermissionsExcept(permissions)]}};
```

Comment:
Redundant spread operator on line 1258 next to `allPermissionsExcept()`. Removing it can improve the performance a little bit.